problems/4-infinite-compression/problem.md

```diff
@@ -1,22 +1,42 @@
 # Infinite avatar compression!
 
-YES BOSS! That's how you do it. Now your minions can do tasks on your behalf. They can even re-delegate to others! It's delegations all the way down.
-
-One thing to note about delegations is that they're not secrets, they can only be used by the target "audience" - in this case the DID of the workshop. So you can embed in scripts and/or send them in emails or whatever really. They're just a proof that you have been given the capability to do something.
-
-Ok, moving on. We talked about content addressing a couple of exercises ago. Right? Yeah well that content addressed data that you uploaded is now a DAG. Yeah, a DAG - Directed Acyclic Graph. So I don't know if you know anything about hashing but the CID you received is no ordinary hash. It's a CID that addresses a graph of _many_ nodes each with their _own_ CID...and with this one very special property - no cycles. Cool huh?! Fun fact: the CID you have is the CID of the root node of the graph. It's the hash of the node's data **and** any links it has to other nodes.
-
-The DAG is packed into a CAR 🚗 - beep beep. jk jk a CAR is a Content-Addressed Archive. It's kinda like a `tar` file but for DAGs. It's a useful container format for transferring content addressed data over HTTP.
-
-In web3.storage, we content address CARs as well! So we generate a hash for the CAR and invoke a `store/add` capability, passing the CAR CID as parameters to the invocation. The service will reply with a signed URL - a place to put the CAR that will only accept data that hashes to the CID of the CAR.
-
-**BUT** if we know the CAR you want to send has already been stored then we'll send back a flag to say that you don't need to send it! Hey presto, infinite compression! Joking aside, isn't content addressing awesome?!
```

Comment:
I feel like we've lost information here.

"Infinite compression" is not real, but is an attempt to put a name to this advantage:

We content address the data, and because we do that, the service can determine if it already has the data without actually sending it. If the service has the data, you do not need to send it - i.e. you have "infinite compression" because you did not send anything!